server/src/main/java/com/linecorp/centraldogma/server/internal/httpapi/RepositoryServiceV1.java

```diff
@@ -0,0 +1,179 @@
+            if (status.get().equalsIgnoreCase("removed")) {
+                return CompletableFuture.completedFuture(
+                        projectManager().get(projectName).repos().listRemoved().stream()
+                                        .map(name -> {
+                                            RepositoryDto r = new RepositoryDto();
+                                            r.setName(name);
+                                            return r;
+                                        }).collect(toList()));
+            } else {
+                // wrong status
+                throw new IllegalArgumentException("invalid status: " + status);
+            }
+        } else {
+            return CompletableFuture.completedFuture(
+                    projectManager().get(projectName).repos().list().values().stream()
+                                    .map(DtoConverter::convert)
+                                    .collect(toList()));
+        }
+    }
+
+    /**
+     * POST /projects/{projectName}/repos
+     *
+     * <p>Creates a new repository.
+     */
+    @Post("/projects/{projectName}/repos")
+    public CompletionStage<RepositoryDto> createRepository(@Param("projectName") String projectName,
+                                                           AggregatedHttpMessage message) {
+        checkProjectExists(projectName);
+
+        final JsonNode jsonNode;
+        try {
+            jsonNode = Jackson.readTree(message.content().toStringAscii()).get("name");
+        } catch (IOException e) {
+            throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+        }
+        checkArgument(jsonNode != null, "repository name should be non-null");
+        final String repoName = jsonNode.asText();
+        checkArgument(!isNullOrEmpty(repoName), "repository name should be non-null");
+
+        return execute(Command.createRepository(System.currentTimeMillis(), AuthenticationUtil.currentAuthor(),
+                                                projectName, repoName))
+                .handle((unused, thrown) -> {
+                    if (thrown != null && thrown.getCause() instanceof StorageExistsException) {
+                        throw new HttpResponseException(HttpStatus.BAD_REQUEST);
+                    }
+                    return DtoConverter.convert(projectManager().get(projectName).repos().get(repoName));
+                });
+    }
+
+    /**
+     * DELETE /projects/{projectName}/repos/{repoName}
+     *
+     * <p>Remove a repository.
+     */
+    @Delete("/projects/{projectName}/repos/{repoName}")
+    public CompletionStage<Object> removeRepository(@Param("projectName") String projectName,
+                                                      @Param("repoName") String repoName) {
+        checkProjectExists(projectName);
+        final long removeTimeMillis = System.currentTimeMillis();
+        return execute(Command.removeRepository(removeTimeMillis, AuthenticationUtil.currentAuthor(),
```

Comment:
You can just use the factory method without timestamp parameter to let the upstream decide.